mapof.go

```diff
@@ -94,6 +95,19 @@ func NewMapOf[K comparable, V any](options ...func(*MapConfig)) *MapOf[K, V] {
 func NewMapOfWithHasher[K comparable, V any](
 	hasher func(K, uint64) uint64,
 	options ...func(*MapConfig),
+) *MapOf[K, V] {
+	cmp := func(a, b K) bool { return a == b }
+	return NewMapOfWithMethods[K, V](hasher, cmp, options...)
```

Comment:
What will happen if someone uses `NewMapOf`/`NewMapOfPresized`/`NewMapOfWithHasher` with a non-comparable type? It looks like now the code will compile, but the default comparator is likely to be the wrong one. If so, do you have any ideas on how to prevent such misuse?